bindings/java/src/org/sleuthkit/datamodel/BlackboardArtifact.java

```diff
@@ -169,19 +170,63 @@ public String getDisplayName() {
 	 * @throws TskCoreException if there is a problem creating the description.
 	 */
 	public String getShortDescription() throws TskCoreException {
-		BlackboardAttribute attr = null;
-		if (artifactTypeId == ARTIFACT_TYPE.TSK_WEB_BOOKMARK.getTypeID()) {
-			attr = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_URL));
+		List<BlackboardAttribute> attrs;
+		attrs = new ArrayList<BlackboardAttribute>();
+		if (artifactTypeId == ARTIFACT_TYPE.TSK_WEB_BOOKMARK.getTypeID()
```

Comment:
I agree a switch would be easier to read but I am having trouble figuring out how to write it since I can't switch on the artifactTypeId because the getTypeId() method isn't a constant expression.